### PR TITLE
Enable tab preloading

### PR DESCRIPTION
**Pull request by zephraph**
Update the `Tabs` component to allow route functionality via the preload link.

### CODE REVIEW

File: src/Styleguide/Components/Tabs.tsx [hunk 1]
```diff
@@ -76,7 +83,10 @@ export class Tabs extends React.Component<TabsProps, TabsState> {
 interface TabProps {
   name: string
 }
-export class Tab extends React.Component<TabProps> {
+interface PreloadLinkTabProps extends TabProps, PreloadLinkProps {
+  children: never
```
Comment (zephraph) [inline src/Styleguide/Components/Tabs.tsx]: This doesn't work. @alloy have any advice here? It seems that typescript isn't using this to check the type of the children.

File: src/Styleguide/Components/Tabs.tsx [hunk 2]
```diff
@@ -49,12 +51,17 @@ export class Tabs extends React.Component<TabsProps, TabsState> {
   }
 
   renderTab = (tab, index) => {
-    const { name } = tab.props
+    const { name, children, ...preloadProps } = tab.props
+    const buttonContent = Object.keys(preloadProps).length ? (
```
Comment (zephraph) [inline src/Styleguide/Components/Tabs.tsx]: Maybe could use more robust logic to see if it's a preload or not.

**Comment by damassi** [comment]
LGTM 👍 - gonna merge and we can iterate.

File: src/Styleguide/Components/Tabs.tsx (hunk 2 above)
Comment (damassi) [reply src/Styleguide/Components/Tabs.tsx]: Can you add a `FIXME` here with this comment? At some point we can check the shape of `preloadProps` against the interface.

**Comment by zephraph** [comment]
This version doesn't work... integrating everything into one component is a bit too much.

**Comment by damassi** [comment]
I've got a naive CSS version going in my branch that piggybacks on router features, when ready i'll go ahead and add back here where appropriate

**Pull request closed without merging**